wrapanapi/containers/providers/rhopenshift.py

```diff
@@ -0,0 +1,874 @@
+            template: Openshift's template object
+            params: bunch of key=value parameters
+        Returns: updated template
+        """
+        template = copy.deepcopy(template)
+        if template.parameters:
+            new_parameters = template.parameters
+            for new_param, new_value in params.items():
+                for index, old_param in enumerate(new_parameters):
+                    if old_param['name'] == new_param:
+                        old_param = new_parameters.pop(index)
+                        if 'generate' in old_param:
+                            old_param['generate'] = None
+                            old_param['_from'] = None
+
+                        old_param['value'] = new_value
+                        new_parameters.append(old_param)
+                        template.parameters = new_parameters
+        return template
+
+    def process_template(self, name, namespace, parameters=None):
+        """Implements template processing mechanizm similar to `oc process`.
+        It does to functions
+          1. parametrized templates have to be processed in order to replace parameters with values.
+          2. templates consist of list of objects. Those objects have to be extracted
+          before creation accordingly.
+
+        Args:
+            name: (str) template name
+            namespace: (str) openshift namespace
+            parameters: parameters and values to replace default ones
+        Return: list of objects stored in template
+        """
+        # workaround for bug https://github.com/openshift/openshift-restclient-python/issues/60
+        raw_response = self.o_api.read_namespaced_template(name=name, namespace=namespace,
+                                                           _preload_content=False)
+        raw_data = json.loads(raw_response.data)
+        updated_data = self.rename_structure(raw_data)
+        read_template = self.ociclient.V1Template(**updated_data)
+        if parameters:
+            updated_template = self.update_template_parameters(template=read_template, **parameters)
+        else:
+            updated_template = read_template
+        raw_response = self.o_api.create_namespaced_processed_template(namespace=namespace,
+                                                                       body=updated_template,
+                                                                       _preload_content=False)
+        raw_data = json.loads(raw_response.data)
+        updated_data = self.rename_structure(raw_data)
+        processed_template = self.ociclient.V1Template(**updated_data)
+        return processed_template.objects
+
+    def rename_structure(self, struct):
+        """Fixes inconsistency in input/output data of openshift python client methods
+
+        Args:
+            struct: data to process and rename
+        Return:
+        """
+        if not isinstance(struct, (str, unicode)) and isinstance(struct, Iterable):
```

Comment:
**REQUIRED**: Use `six_string_types` or how is it called exactly. This is not py3 compatible.